docs/recipes/build/add-build-script.md

```diff
@@ -1,88 +1,120 @@
 ```fsharp
-#r "paket: groupref build //"
-#load "./.fake/build.fsx/intellisense.fsx"
-#r "netstandard"
-
 open Fake.Core
-open Fake.Core.TargetOperators
-open Fake.DotNet
 open Fake.IO
+open System
 
-Target.initEnvironment ()
+let redirect createProcess =
+    createProcess
+    |> CreateProcess.redirectOutputIfNotRedirected
+    |> CreateProcess.withOutputEvents Console.WriteLine Console.WriteLine
 
-let serverPath = Path.getFullName "./src/Server"
-let clientPath = Path.getFullName "./src/Client"
-let deployDir = Path.getFullName "./deploy"
+let createProcess exe arg dir =
+    CreateProcess.fromRawCommandLine exe arg
+    |> CreateProcess.withWorkingDirectory dir
+    |> CreateProcess.ensureExitCode
 
-let npm args workingDir =
-    let npmPath = ProcessUtils.tryFindFileOnPath "npm" |> Option.get
-    let arguments = args |> String.split ' ' |> Arguments.OfArgs
+let dotnet = createProcess "dotnet"
 
-    Command.RawCommand (npmPath, arguments)
-    |> CreateProcess.fromCommand
-    |> CreateProcess.withWorkingDirectory workingDir
-    |> CreateProcess.ensureExitCode
+let npm =
+    let npmPath =
+        match ProcessUtils.tryFindFileOnPath "npm" with
+        | Some path -> path
+        | None -> failwith "npm was not found in path."
+    createProcess npmPath
+
+let run proc arg dir =
+    proc arg dir
     |> Proc.run
     |> ignore
 
-let dotnet cmd workingDir = DotNet.exec (DotNet.Options.withWorkingDirectory workingDir) cmd ""
+let execContext = Context.FakeExecutionContext.Create false "build.fsx" [ ]
+Context.setExecutionContext (Context.RuntimeContext.Fake execContext)
+
+Target.create "Clean" (fun _ -> Shell.cleanDir (Path.getFullName "deploy"))
+
+Target.create "InstallClient" (fun _ -> run npm "install" ".")
 
-Target.create "Bundle" (fun _ ->
-    Shell.cleanDir deployDir
-    npm "install" clientPath
-    dotnet (sprintf "publish -c Release -o \"%s\"" deployDir) serverPath
-    npm "run build" clientPath
+Target.create "Run" (fun _ ->
+    run dotnet "build" (Path.getFullName "src/Shared")
+    [ dotnet "watch run" (Path.getFullName "src/Server")
+      dotnet "fable watch --run webpack-dev-server" (Path.getFullName "src/Client") ]
+    |> Seq.toArray
+    |> Array.map redirect
+    |> Array.Parallel.map Proc.run
+    |> ignore
 )
 
-"Clean"
-    ==> "InstallClient"
-    ==> "Bundle"
+open Fake.Core.TargetOperators
 
-Target.runOrDefaultWithArguments "Bundle"
+let dependencies = [
+    "Clean"
+        ==> "InstallClient"
+        ==> "Run"
+]
+
+[<EntryPoint>]
+let main args =
+  try
+      match args with
+      | [| target |] -> Target.runOrDefault target
+      | _ -> Target.runOrDefault "Run"
+      0
+  with e ->
+      printfn "%A" e
+      1
 ```
 
-#### 4. Build the Application
-You can now build your application by running the following command at the root directory of your solution:
-```powershell
-dotnet fake build
-```
\ No newline at end of file
+#### 3. Move the files
```

Comment:
At the start of this recipe it says to create the project at the root; here it seems to now suggest that you have to move it anyway?

Reply:
When you make a new project from the command line, it gets put into a folder.

It was that or get them to new up each file by hand which I didn't fancy.

Reply:
What about ```dotnet new console -lang f# -n Build -o .```?

Reply:
Yeah, that does do it. 

I did think that the explicit step of moving them to the root gave me a good chance to explain 'This is weird but we are doing it to allow you to run the build cmd at the root'.

I could add that as a comment anyway though.

Reply:
Done